tests/models.py

```diff
@@ -20,25 +21,75 @@ def lt40(self):
         return self.filter(amount__lt=40)
 
 
-class LT40Manager(CTEManager):
+class LT30QuerySet(CTEQuerySet):
+
+    def lt30(self):
+        return self.filter(amount__lt=30)
+
+
+class LT25QuerySet(CTEQuerySet):
+
+    def lt25(self):
+        return self.filter(amount__lt=25)
+
 
+class LT40Manager(CTEManager):
+    # This is a semi broke manager but allows testing of the documented
+    # use of get_queryset. It is broken by ignoring the queryset passed
+    # to it via the static method from_queryset
```

Comment:
What do you think of removing this broken thing and fixing (or removing) the test so it does not rely on it? If it requires a backward-incompatible change we can push a major version and document it, along with a suggestion for how to fix broken code, in the change log.

Reply:
I had added the check to see if i introduced any backward-incompatible changes.

I think we should document the 'new' way and simpler way but the old will still work but didn't want to change the manager and remove test for the documented manger setup in the same commit.